sandbox/modules/identity/src/main/java/org/opensearch/identity/rest/service/PermissionService.java

```diff
@@ -0,0 +1,78 @@
+/*
+* SPDX-License-Identifier: Apache-2.0
+*
+* The OpenSearch Contributors require contributions made to
+* this file be licensed under the Apache-2.0 license or a
+* compatible open source license.
+*/
+
+package org.opensearch.identity.rest.service;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.ActionListener;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.identity.ConfigConstants;
+import org.opensearch.identity.authz.OpenSearchPermission;
+import org.opensearch.identity.authz.PermissionFactory;
+import org.opensearch.identity.authz.PermissionStorage;
+import org.opensearch.identity.rest.action.permission.put.PutPermissionResponse;
+import org.opensearch.identity.rest.action.permission.put.PutPermissionResponseInfo;
+import org.opensearch.index.IndexNotFoundException;
+import org.opensearch.transport.TransportService;
+import org.opensearch.identity.utils.ErrorType;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.unmodifiableList;
+
+/**
+ * Service class for Permission related functions
+ */
+@SuppressWarnings("unchecked")
+public class PermissionService {
+
+    private static final Logger logger = LogManager.getLogger(PermissionService.class);
+
+    private final ClusterService clusterService;
+    private final TransportService transportService;
+    private final NodeClient nodeClient;
+
+    private final String identityIndex = ConfigConstants.IDENTITY_CONFIG_INDEX_NAME;
+
+    @Inject
+    public PermissionService(ClusterService clusterService, TransportService transportService, NodeClient nodeClient) {
+        this.clusterService = clusterService;
+        this.transportService = transportService;
+        this.nodeClient = nodeClient;
+    }
+
```

Comment:
```suggestion
    private final String identityIndex;

    @Inject
    public PermissionService(Settings settings, ClusterService clusterService, TransportService transportService, NodeClient nodeClient) {
        this.clusterService = clusterService;
        this.transportService = transportService;
        this.nodeClient = nodeClient;
        this.identityIndex = settings.get(
            IdentityConfigConstants.IDENTITY_CONFIG_INDEX_NAME,
            IdentityConfigConstants.IDENTITY_DEFAULT_CONFIG_INDEX
        );
    }

```

this would solve the tests failure `Identity index not initialized`

Reply:
Thanks for the tip!